CMSgov/easi-app
# [NOREF] useMessage alert updates

# NOREF

## Changes and Description

This fixes a bug where assigning a TRB lead from the admin homepage showed two Alerts nested inside of each other:

![screenshot_2023-06-26_at_10 23 54_am](https://github.com/CMSgov/easi-app/assets/60187543/390054c5-beb7-479a-a8aa-47a0c895839c)

It also solves some inconsistencies when using the `useMessage` hook that has caused a couple of different bugs. We currently only use it for displaying alerts, but there are two different ways we've been using it:
1. Pass a string to `showMessage` and display `message` within an `Alert` component
2. Pass an `Alert` component to `showMessage` and then display `message` with no wrapper

This caused issues when `message` is an `Alert` component but the page displaying the message wrapped it in an additional `Alert` component.

To prevent those inconsistencies, I updated the hook to specifically display our Alert component:
- `showMessage` and `showMessageOnNextPage` now accept an object containing `Alert` component props and a `message` prop that can be a string or react element
- useMessage now returns a `<Message />` component that displays the message prop wrapped in an `Alert` with the provided props
  - The <Message /> component accepts a `className` prop that overwrites the props provided to `showMessage`. This is helpful if certain pages need different classes/styling when displaying the same message


**Usage example:**
```
// Initialize hook
const { Message, showMessage, showMessageOnNextPage } = useMessage();

// Setting message
showMessage({
  message: 'Message text here',
  type: 'success'
  // Optional Alert props can be added here
});

// Displaying message
<Message className="optional-class" />
```

Message component from above example returns this Alert:
```
<Alert type="success" className="optional-class">
  Message text here
</Alert>
```



## How to test this change

To test that the assign TRB lead bug is fixed:
1. Sign in as TRB admin
2. In the "New requests" table on the admin homepage, click "Change lead" on one of the requests
3. Assign a new TRB lead
4. Verify that success message shows as expected

## PR Author Review Checklist



## PR Reviewer Guidelines
- It's best to pull the branch locally and test it, rather than just looking at the code online!
- Check that all code is adequately covered by tests - if it isn't feel free to suggest the addition of tests.
- Always make comments, even if it's minor, or if you don't understand why something was done.
